### PR TITLE
Implement MutinyEmitter for Mutiny types

### LINKED ISSUE #710: Reactive messaging Emitter<T> interface is missing Mutiny integration
`Quarkus version: 1.7.0`

`package org.eclipse.microprofile.reactive.messaging;`

Why does the Emitter interface has only the `CompletionStage<Void> send(T msg)` method and is missing `Uni<Void>` send overload?

As shown here I can consume the messages using mutiny types https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2/emitter/emitter.html#streams

`@Inject @Channel("my-channel") Multi<String> streamOfPayloads;`

But I can only send them using the Emitter that doesn't understand mutiny

`CompletionStage<Void> send(T msg)`

Does it make sense to have something like `Uni<Void> send(T msg);` and be consistent with the consuming interface?

### PR DESCRIPTION
**Pull request by kenfinnigan**
Fixes #710

### CODE REVIEW

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/EmitterConfiguration.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
  */
 public class EmitterConfiguration {
     public String name;
+    public Boolean isMutinyEmitter;
```
Comment (cescoffier) [inline smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/EmitterConfiguration.java:14]: why Object? Isn't `boolean` enough?

**Review by cescoffier — CHANGES_REQUESTED** [review]
LGTM, Can you update the documentation?

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/EmitterConfiguration.java (hunk 1 above)
Comment (kenfinnigan) [reply smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/EmitterConfiguration.java:14]: Nuts, missed that one

**Comment by kenfinnigan** [comment]
Sorry blanked on the docs!

**Comment by kenfinnigan** [comment]
Should be sorted now, sorry about that!

**Review by cescoffier — APPROVED** [review]
Thanks!

**Pull request merged**